### PR TITLE
Update pulumi-kubernetes workflows to use kind and free up disk space

### PR DESCRIPTION
**Pull request by rquitales**
Replicates changes in:

- https://github.com/pulumi/pulumi-kubernetes/pull/2875
- https://github.com/pulumi/pulumi-kubernetes/pull/2831

Fixes: https://github.com/pulumi/pulumi-kubernetes/issues/2878

### CODE REVIEW

**Comment by rquitales** [comment]
Ah, need to update the TypeScript source, not the generated workflow.

File: native-provider-ci/providers/kubernetes/repo/.github/workflows/build.yml [hunk 1]
```diff
@@ -419,6 +419,8 @@ jobs:
     - name: Set PreRelease Version
       run: echo "GORELEASER_CURRENT_TAG=v$(pulumictl get version --language generic)"
         >> $GITHUB_ENV
+    - name: Free Disk Space (Ubuntu)
+      uses: jlumbroso/free-disk-space@main
```
Comment (mjeffryes) [inline native-provider-ci/providers/kubernetes/repo/.github/workflows/build.yml:423]: note this is already in the workflow at line 394. (I added to ci-mgmt, and was just going to wait for it to propagate.)

**Review by mjeffryes — APPROVED** [review]
Please just ship the kind changes, since the disk space work is already in ci-mgmt.

**Comment by rquitales** [comment]
> Please just ship the kind changes, since the disk space work is already in ci-mgmt.

I've added a hotfix for the disk space work in this PR. The build job is still failing in kubernetes (and other native providers that use ci-mgmt as well) since the `jlumbroso/free-disk-space` action does not support major version tagging. This PR fixes the action to the latest release.

**Pull request merged**